src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionMidEventHandler.java

```diff
@@ -0,0 +1,79 @@
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.IssueDocumentHandler;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.config.DocumentConfiguration;
+import uk.gov.hmcts.reform.sscs.docassembly.GenerateFile;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class AdminActionCorrectionMidEventHandler extends IssueDocumentHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+    private final DocumentConfiguration documentConfiguration;
+    private final GenerateFile generateFile;
+    @Value("${feature.postHearings.enabled}")
+    private final boolean isPostHearingsEnabled;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.MID_EVENT)
+            && callback.getEvent() == EventType.ADMIN_ACTION_CORRECTION
+            && isPostHearingsEnabled;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+        String userAuthorisation) {
+
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+        String pageId = callback.getPageId();
+        String caseId = sscsCaseData.getCcdCaseId();
+        log.info("Admin Action Correction: handling callback with pageId {} for caseId {}", pageId, caseId);
+
+        PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse =
+            new PreSubmitCallbackResponse<>(sscsCaseData);
+        AdminCorrectionType adminCorrectionType = sscsCaseData.getPostHearing().getCorrection().getAdminCorrectionType();
+
+        if (isNull(adminCorrectionType)) {
+            log.error(String.format("adminCorrectionType unexpectedly null for case: %s", caseId));
+            preSubmitCallbackResponse.addError(String.format("adminCorrectionType unexpectedly null for case: %s", caseId));
+
+        } else if (AdminCorrectionType.HEADER.equals(adminCorrectionType)) {
+            log.info("Handling header correction for case: {}", caseId);
+            YesNo noticeGenerated = sscsCaseData.getFinalDecisionNoticeGenerated();
+
+            if (isYes(noticeGenerated)) {
+                // IF generated: automatically regenerate final decision with the current details
+                log.info("Admin Action Correction: Regenerating final decision for caseId {}", caseId);
+                String templateId = documentConfiguration.getDocuments()
+                    .get(sscsCaseData.getLanguagePreference()).get(DECISION_ISSUED);
+                preSubmitCallbackResponse = issueDocument(callback, DECISION_NOTICE, templateId, generateFile, userAuthorisation);
+            } else {
+                // IF uploaded: go to upload screen and expect user to upload new document
```

Comment:
should that be done? if not and something will be added here as part of other ticket add TODO

Reply:
I'm not sure if there is any code required here to instigate this functionality, or if it is entirely caused by ccd defs. Will add TODO to that effect anyway.